test/e2e/plain_provisioner_test.go

```diff
@@ -345,6 +345,120 @@ var _ = Describe("plain provisioner bundle", func() {
 			))
 		})
 	})
+
+	When("Bundles are backed by a git repository", func() {
+		var (
+			bundles []*rukpakv1alpha1.Bundle
+			ctx     context.Context
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the git based Bundles")
+			bundles = []*rukpakv1alpha1.Bundle{
```

Comment:
I think my preference would be to split out each of these Bundle resources into their own nested When containers, e.g. When("a Bundle contains a reference to a git repository tag"), and handle any of the requisite checks within those individual test containers. Thoughts?

Reply:
We can hold off on fixing this in this initial implementation but I'd like to see us refactor these e2e test cases in a follow-up.